src/core/automount.c

```diff
@@ -574,6 +577,22 @@ static void automount_enter_waiting(Automount *a) {
         if (r < 0)
                 goto fail;
 
+        r = fstab_filter_options(
+                a->options,
+                "fd\0"
+                "pgrp\0"
+                "minproto\0"
+                "maxproto\0"
+                "direct\0"
+                "indirect\0",
+                NULL,
+                NULL,
+                NULL,
+                &user_options);
+        if (r < 0)
+                goto fail;
```

Comment:
quite frankly, if you filter it I think it would make sense to explicitly fail the operation if any of the options we use ourselves are specified.

Reply:
I was trying to keep the behavior consistent with [mount units](https://github.com/systemd/systemd/blob/b52b5763e733a0b7cf73509c5467923035e22596/src/core/mount.c#L1043).

If renamed to ExtraOptions, then explicit failing probably makes sense.

Reply:
those three options we strip in mount units are userspace option that we already process in the fstab generator thing, kernel isn't supposed to see them...